### PR TITLE
fix: support `SequenceState, SimpleInterval`

### PR DESCRIPTION
**Pull request by aguliyev**
fix: support `SequenceState` with `LiteralSequenceExpression` and `SimpleInterval` with `SequenceInterval`

This PR is main->main

There is similar patch on tag 0.7.4 :
https://github.com/ga4gh/vrs-python/compare/0.7.4...aguliyev:SIPAW-284_Add_support?expand=1

### CODE REVIEW

**Comment by aguliyev** [comment]
@andreasprlic , my concern is that VDB uses 0.7.4 , not the latest `main`
so applying this PR is fine, but we need to also update 0.7.4. 
Or, do you mean we update in VDB to use the latest code?

**Comment by andreasprlic** [comment]
@aguliyev 0.7.4 is currently the latest stable release. I believe the next to follow would be 0.7.5. We would switch over to that.

**Comment by korikuzma** [comment]
Instead of allowing both should we instead translate deprecated to non-deprecated models as noted in [this](https://github.com/ga4gh/vrs-python/issues/72) issue?

**Comment by aguliyev** [comment]
according to https://github.com/ga4gh/vrs-python#vrs-python-and-vrs-version-correspondence
there should be

> vrs-python 0.7 branch tracks the vrs 1.2 branch.

but checking the list of branches I do not see it

**Comment by andreasprlic** [comment]
@[korikuzma](https://github.com/korikuzma) can you explain what is meant with "translate" between the two models? What should the code do here?

**Comment by korikuzma** [comment]
> @[korikuzma](https://github.com/korikuzma) can you explain what is meant with "translate" between the two models? What should the code do here?

So if the user inputs an Allele using a SequenceState (deprecated), we would convert it into a LiteralSequenceExpression. Similarly, if the user inputs a SimpleInterval (deprecated), we would convert it into a SequenceInterval.

**Comment by aguliyev** [comment]
> should we instead translate deprecated to non-deprecated models

normally yes; But that will lead to different model, hence different generated identifier(hash).
That is the main reason of us making this fix - we want to continue generating allele VRS hashes from the old model.
(the hash is based on the model serialization value)

**Comment by andreasprlic** [comment]
@[korikuzma](https://github.com/korikuzma) what do you think of the new `ensure_allele_is_latest_model`?

**Comment by korikuzma** [comment]
> @[korikuzma](https://github.com/korikuzma) what do you think of the new `ensure_allele_is_latest_model`?

I like it! @aguliyev raises a good point about people still using deprecated classes. My initial thought is to set a bool in the translate to/from methods on whether or not to allow deprecated classes, where the default is set to only allow or translate to the latest. @andreasprlic , what do you think?

File: src/ga4gh/vrs/extras/translator.py
Comment (korikuzma) [inline src/ga4gh/vrs/extras/translator.py:365]:
```suggestion
            raise ValueError("_to_hgvs requires a VRS Allele with SequenceLocation and LiteralSequenceExpression")
```

File: src/ga4gh/vrs/extras/translator.py
Comment (korikuzma) [inline src/ga4gh/vrs/extras/translator.py:452]:
```suggestion
            raise ValueError("_to_spdi requires a VRS Allele with SequenceLocation and LiteralSequenceExpression")
```

**Review by korikuzma — COMMENTED** [review]
This may change depending on the codeowners' opinions on my [latest comment](https://github.com/ga4gh/vrs-python/pull/86#issuecomment-1090873859).

File: src/ga4gh/vrs/extras/translator.py
Comment (aguliyev) [reply src/ga4gh/vrs/extras/translator.py]: ok

File: src/ga4gh/vrs/extras/translator.py
Comment (aguliyev) [reply src/ga4gh/vrs/extras/translator.py]: ok

**Review by andreasprlic — APPROVED** [review]
LGTM

**Comment by andreasprlic** [comment]
I feel the translate_to methods should be generally safe to use update allele models (that have deprecated mapped to current) models. As long as we are not calculating any hashes, there should be no risk here.

**Comment by aguliyev** [comment]
I dont have merge permissions.
Plz, could somebody merge;  also, should we make tag/release? Because we need to use it in our project...

**Comment by andreasprlic** [comment]
Once this is merged, what will it take to get this out in a release?

**Comment by aguliyev** [comment]
another question - is the current `main` which we patch here a production ready?  (checking git tags, looks like it is dev/alpha).
Should we make a branch off tag 0.7.4 and patch that one also?

FYI @andreasprlic

**Comment by ahwagner** [comment]
@aguliyev this was my concern too; as you pointed out we never made a `0.7` branch, and IIRC `main` currently has some `0.8` dev work in it. I plan to review and clean that up tonight.

As for versioning, @jsstevenson added an [auto-deploy action to GH](https://github.com/ga4gh/vrs-python/blob/main/.github/workflows/release.yml). We just have to version tag a commit on `main` and it will build and deploy to PyPI. Any of the VRS devs should be able to do this.

**Comment by andreasprlic** [comment]
@ahwagner could we just create a new branch off the latest 0.7.X release tag, merge this PR into that branch (as well as main), apply a new tag, and this would give us a new release in that series?

**Comment by ahwagner** [comment]
@andreasprlic and @aguliyev this is now live on PyPI as `0.7.5`.

**Comment by andreasprlic** [comment]
Great, thank you @ahwagner !

**Comment by aguliyev** [comment]
thank you guys.
plz, notice that now 0.7.5 becomes newer than 0.8.x
so, 0.7.5 is no longer tied to vrs 1.2 branch (as it is told in README.
Also, do we want to make a release?

**Pull request merged**
